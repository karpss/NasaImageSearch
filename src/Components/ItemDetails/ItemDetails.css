.backButton{
    padding: 10px 16px;
    margin: 10px 0;
     border-radius: 5px;
      cursor: pointer;
    outline: none;
}

.cardDetails{
    display: inline-block;
    background: #fff;
    padding: 1em;
    margin: 0 0 1.5em;
    width: 50%;
    box-sizing: border-box;
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
    box-shadow: 2px 2px 4px 0 #ccc;
}

.cardDetails img{
    width:100%;
 }