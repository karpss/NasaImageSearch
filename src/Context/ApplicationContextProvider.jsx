/* eslint-disable */
import  {createContext, useReducer } from "react";

export const ApplicationContext = createContext({});

export const ActionTypes = {
  FETCH_IMAGE_ITEMS : "FETCH_ITEMS",
  FETCH_IMAGE_ITEMS_FULFILLED : "FETCH_ITEMS_FULFILLED",
  FETCH_IMAGE_ITEMS_REJECTED : "FETCH_ITEMS_REJECTED",
  SELECT_ITEM : "SELECT_ITEM"
}


const INITIAL_STATE = {
  items: [],
  selectedItem: null,
  loading: false,
  error: null
};

const reducer = (state, action) => {
  console.log(action.type);
  switch (action.type) {
    case ActionTypes.FETCH_IMAGE_ITEMS:
      return { ...state, loading: true };
    case ActionTypes.FETCH_IMAGE_ITEMS_FULFILLED:
      return { ...state, items: action.payload, loading: false };
    case ActionTypes.SELECT_ITEM:
      return { ...state, selectedItem: action.payload };
    case ActionTypes.FETCH_IMAGE_ITEMS_REJECTED:
      return { ...state, loading: false, error: "action.payload" };
      default:
        return state;
      
  }
  
  
  
};


const ApplicationContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);
  return (
    <ApplicationContext.Provider value={{ state, dispatch }}>
      {children}
    </ApplicationContext.Provider>
  );
};

export default ApplicationContextProvider;
