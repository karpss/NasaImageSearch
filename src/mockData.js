/* eslint-disable */
export const mockData = {
    
    collection:{
        
        
            
            "href": "https://images-assets.nasa.gov/image/PIA12235/collection.json",
            
            "data": [
                {
                    "center": "JPL",
                    "media_type": "image",
                    "description": "Nearside of the Moon",
                    "nasa_id": "PIA12235",
                    "photographer": "NASA/Bill Ingalls",
                    "keywords": [
                        "Moon",
                        "Chandrayaan-1"
                    ],
                    "location": "Mars, PA, USA",
                    "title": "Nearside of the Moon",
                    "date_created": "2009-09-24T18:00:22Z"
                }
            ],
            "links": [
                {
                    "href": "https://images-assets.nasa.gov/image/PIA12235/PIA12235~thumb.jpg",
                    "render": "image",
                    "rel": "preview"
                }
            ],
            
        
    }   

}
