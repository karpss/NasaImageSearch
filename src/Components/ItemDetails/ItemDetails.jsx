import { useContext, useEffect, useState } from "react";
import {
  ActionTypes,
  ApplicationContext,
} from "../../Context/ApplicationContextProvider";
import "./ItemDetails.css";

function ItemDetails() {
  const [itemMedia, setItemMedia] = useState({
    images: [],
  });

  const {
    state: { selectedItem },
    dispatch,
  } = useContext(ApplicationContext);

  useEffect(() => {
    if (selectedItem) {
      fetch(
        `https://images-api.nasa.gov/asset/${selectedItem?.data[0].nasa_id}`,
      )
        .then((res) => res.json())
        .then((data) => {
          setItemMedia({
            images: data.collection.items.filter((i) =>
              i.href.endsWith(".jpg"),
            ),
          });
        })
        .catch((error) => {
          return error;
        });
    }
  }, [selectedItem, dispatch]);

  return (
    <div className="cardDetails">
      <div>
        <button
          className="backButton"
          type="button"
          onClick={() => {
            dispatch({ type: ActionTypes.SELECT_ITEM, payload: null });
          }}
        >
          Back to search
        </button>
      </div>
      <div>
        <img src={itemMedia.images[0]?.href} alt="Nasa Gallery" />
      </div>
      <div>
        <h2> Title: {selectedItem.data[0].title}</h2>
        <b> Date: {new Date(selectedItem.data[0].date_created).toString()} </b>
        <p>Center: {selectedItem.data[0].center}</p>
        <p> Description: {selectedItem.data[0].description}</p>
        <p>Location: {selectedItem.data[0].location}</p>
        <p>Photographer: {selectedItem.data[0].photographer}</p>
        <p>Keywords: {selectedItem.data[0].keywords.join(", ")}</p>
      </div>
    </div>
  );
}

export default ItemDetails;
