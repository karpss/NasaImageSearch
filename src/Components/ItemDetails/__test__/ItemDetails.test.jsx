/* eslint-disable */
import { render} from "@testing-library/react";
import { mockData } from "../../../mockData";
import ItemDetails from "../ItemDetails";
import ApplicationContextProvider from "../../../Context/ApplicationContextProvider";


test("matches snapshot", () => {
    const { container } = render(
        <ApplicationContextProvider> 
    <ItemDetails {...mockData} />
    </ApplicationContextProvider>
    
    );
    expect(container.firstChild).toMatchSnapshot();
  });