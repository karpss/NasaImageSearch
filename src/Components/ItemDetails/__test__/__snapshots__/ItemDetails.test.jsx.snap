// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="cardDetails"
>
  <div>
    <button
      class="backButton"
      type="button"
    >
      Back to search
    </button>
  </div>
  <div>
    <img
      alt="Nasa Gallery"
    />
  </div>
  <div>
    <h2>
       Title: 
    </h2>
    <b>
       Date: 
      Invalid Date
       
    </b>
    <p>
      Center: 
    </p>
    <p>
       Description: 
    </p>
    <p>
      Location: 
    </p>
    <p>
      Photographer: 
    </p>
    <p>
      Keywords: 
    </p>
  </div>
</div>
`;
